src/main/java/io/stargate/sgv2/jsonapi/service/shredding/Shredder.java

```diff
@@ -214,22 +215,40 @@ private void validateDocument(DocumentLimitsConfig limits, ObjectNode doc, Strin
     }
 
     // Second: traverse to check for other constraints
-    validateObjectValue(limits, null, doc, 0);
+    AtomicInteger totalProperties = new AtomicInteger(0);
```

Comment:
Reason of usinh AtomicInteger here is **just** for wrapping the counter, and enabling to change it in method calls?
Is that correct?

Reply:
Yes, exactly. Could use `int[1]` as an alternative but `AtomicInteger` seems better (even if giving wrong idea of needing atomicity).